tasks/resource.json

```diff
@@ -0,0 +1,18 @@
+{
+  "description": "Apply a single Puppet resource",
+  "input_method": "stdin",
+  "parameters": {
+    "type": {
+      "description": "The type of resource to apply",
+      "type": "String[1]"
+    },
+    "title": {
+      "description": "The title of the resource to apply",
+      "type": "String[1]"
```

Comment:
it occurred to me that depending on the parameters you use title may not be useful. Should make this optional and use a default if it's not supplied or is it worth requiring this to get better error messages?

Reply:
Generally there's a 1-to-1 correspondence between title and some required parameter of the resource. So I think if you would otherwise specify that parameter and _not_ title, then you should instead specify title and not that parameter.

I don't think we can really know when a title would be required vs. not, and it could be really bad if we injected a made-up title in the wrong place.

Reply:
I guess the usecase for not using the title is when you're trying to avoid conflict which doesn't apply here.  Parameters should probably be optional though so you can just supply the title.